meson.build

```diff
@@ -347,37 +347,20 @@ libdl_dep = cxx.find_library('dl') # DL library
 thread_dep = dependency('threads') # pthread for tensorflow-lite
 
 iniparser_dep = dependency('iniparser', required : false, version : '>=4.1') # iniparser
-if get_option('platform') == 'android'
-  message('preparing iniparser')
-  run_command(meson.source_root() / 'jni' / 'prepare_iniparser.sh', meson.build_root(), check: true)
-  iniparser_root = meson.build_root() / 'iniparser'
-  iniparser_dep = declare_dependency(include_directories: [ 'iniparser/src' ])
-endif
 
 if not iniparser_dep.found()
-  message('falling back to find libiniparser library and header files')
-  libiniparser_dep = cxx.find_library('iniparser')
-  sysroot = run_command(
-    cxx.cmd_array() + ['-print-sysroot']
-    ).stdout().split('\n')[0]
-
-  if sysroot.startswith('/')
-    sysroot_inc_cflags_template = '-I@0@/usr/include@1@'
-    sysroot_inc = sysroot_inc_cflags_template.format(sysroot, '')
-    add_project_arguments(sysroot_inc, language: ['c', 'cpp'])
-    sysroot_inc_cflags_iniparser = sysroot_inc_cflags_template.format(sysroot,
-      '/iniparser')
-  else
-    sysroot_inc_cflags_iniparser = '-I/usr/include/iniparser'
-  endif
-
-  if libiniparser_dep.found() and cxx.has_header('iniparser.h', \
-        args : sysroot_inc_cflags_iniparser)
-    iniparser_dep = declare_dependency (dependencies : libiniparser_dep,
-      compile_args : sysroot_inc_cflags_iniparser)
-  else
-    error('Failed to resolve dependency on iniparser')
-  endif
+  # CMake module
+  cmake = import('cmake')
+
+  # iniparser
+  iniparser_root = meson.source_root() / 'subprojects' / 'iniparser'
+  iniparser_options = cmake.subproject_options()
+  iniparser_options.add_cmake_defines({'BUILD_DOCS': false})
+  iniparser_options.add_cmake_defines({'BUILD_EXAMPLES': false})
+  iniparser_options.add_cmake_defines({'BUILD_TESTING': false})
+  iniparser_options.add_cmake_defines({'BUILD_SHARED_LIBS': false})
+  iniparser_subproject = cmake.subproject('iniparser', options: iniparser_options, required: true)
+  iniparser_dep = iniparser_subproject.dependency('iniparser-static')
```

Comment:
pdebuild (Ubuntu platform build) and gbs/obs (Tizen platform build) does NOT allow referencing external repo. We need to keep the currentfallback mechanism because Ubuntu-iniparser does not have pkgconfig support and we still need to link to such packages.

If the platform (Ubuntu, Tizen, and so on) has the iniparser, we need to link to the platform-supplied shared library.
So, please restrict the usage of external repos to the case that really needs it.

(this is not related to meson versioning issue.)